license_manager/server/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"flag"
+	"net"
+	"strconv"
+
+	lmpb "github.com/enfabrica/enkit/license_manager/proto"
+	"github.com/enfabrica/enkit/license_manager/service"
+
+	"github.com/golang/glog"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+var (
+	port = flag.Int("port", 8080, "Port for gRPC services")
+)
+
+func main (){
+	flag.Parse()
+
+	addr := net.JoinHostPort("", strconv.FormatInt(int64(*port), 10))
+	listen, err := net.Listen("tcp", addr)
+	if err != nil {
+		glog.Fatalf("Failed to listen on %q: %v", addr, err)
+	}
+	server := grpc.NewServer()
```

Comment:
FYI (don't change anything): if we were planning to run this on google app engine or outside our network, I'd recommend using our ugly lib/server lib/client, and grow them. Reason: the libraries wrap grpc around auth and grpcweb, so we can actually run apps behind non-http2 capable proxies, and can create UIs via piggybacking on grpcweb.

Why I'd leave this as is: we have no reasonable way to run the real license manager in the cloud at this point in time. The protocol is extremely fragile at best, has no authentication, the server is proprietary code that is supposed to require baremetal machines, and hard codes the MAC address in the licenses supplied by the vendor.

So, there is no need for our internal wrappers, and by using the native libraries, your code brings in fewer dependencies that can break.

If we were planning to implement a statusz, or were planning to run this in cloud run or app engine, I'd recommend using the wrappers instead.

Reply:
Thanks for the suggestion; using the wrapper does lighten this main up quite a bit, eliminates the flag and solves the logging problem. I'll go with that until a change to our wrapper breaks this.